secrets/awskms/kms.go

```diff
@@ -98,9 +99,28 @@ func (k *keeper) ErrorAs(err error, i interface{}) bool {
 }
 
 // ErrorCode implements driver.ErrorCode.
-func (k *keeper) ErrorCode(error) gcerrors.ErrorCode {
-	// TODO(shantuo): try to classify aws error codes
-	return gcerrors.Unknown
+func (k *keeper) ErrorCode(err error) gcerrors.ErrorCode {
+	ae, ok := err.(awserr.Error)
+	if !ok {
+		return gcerr.Unknown
+	}
+	ec, ok := errorCodeMap[ae.Code()]
+	if !ok {
+		return gcerr.Unknown
+	}
+	return ec
+}
+
+var errorCodeMap = map[string]gcerrors.ErrorCode{
+	kms.ErrCodeNotFoundException:          gcerrors.NotFound,
+	kms.ErrCodeDisabledException:          gcerrors.PermissionDenied,
+	kms.ErrCodeInvalidCiphertextException: gcerrors.InvalidArgument,
+	kms.ErrCodeKeyUnavailableException:    gcerrors.NotFound,
+	kms.ErrCodeDependencyTimeoutException: gcerrors.DeadlineExceeded,
+	kms.ErrCodeInvalidGrantTokenException: gcerrors.PermissionDenied,
+	kms.ErrCodeInternalException:          gcerrors.Internal,
+	kms.ErrCodeInvalidStateException:      gcerrors.InvalidArgument,
```

Comment:
This one sounds more like FailedPrecondition.  It's not that the request was bad, it just assumed something about the system that isn't true.

Reply:
Done.